[ADD] missing strings.h header file is added to src/yaml_parser.c

KaanaryOverFlow opened the pull request:
## 🔒 Security Analysis PR Description

While compiling checker_tools on linux system, the compiler throws an error for `strcasecmp` function in `src/yaml_parser.c`. As provied in man page of strcasecmp, adding
```c
#include <strings.h>
```
into src/yaml_parser.c would fix it.

**Security Impact**: None. it is an compile error<!-- Describe the security implications of this change -->
**Vulnerability Type**: None. same as Security Impact<!-- If applicable, specify the type of vulnerability being addressed (CVE, crypto weakness, dependency issue, etc.) -->
**Languages Affected**: C<!-- List programming languages affected (C/C++, JavaScript/TypeScript, Python, Rust, Go, Dart) -->

---

## ✅ Security Analysis Checklist

- [X] Code follows MasterFabric security standards and GNU coding guidelines
- [ ] Security tests are written and all tests are passing (`make test`)
- [ ] No compiler warnings (`make clean && make`)
- [ ] Security policy compliance verified
- [ ] CVE database integration tested (if applicable)
- [ ] Real-time vulnerability scanning tested (if applicable)
- [ ] Binary hardening checks verified (if applicable)
- [ ] Cryptographic analysis validated (if applicable)
- [ ] Static analysis integration tested (if applicable)
- [ ] Documentation updated for security features
- [ ] Demo applications tested with new security checks
- [X] The PR has been reviewed by at least one security team member

---

## 🛡️ Security Testing Steps

<!-- Provide clear steps to test the security analysis changes, including how to reproduce vulnerabilities or verify fixes. -->

### Prerequisites
1. Build the security checker: `cd checker_tool && make`
2. Install dependencies: `brew install curl json-c pkg-config` (macOS) or `sudo apt-get install libcurl4-openssl-dev libjson-c-dev pkg-config` (Ubuntu)

### Testing Commands
1. **Basic Security Check**:
   ```bash
   ./checker_tool/bin/masterfabric-prerelease --check cases/demo_vulnerable_app --verbose
   ```

2. **Flutter Security Analysis**:
   ```bash
   ./checker_tool/bin/masterfabric-prerelease --check cases/demo_flutter_vulnerable_app --verbose
   ```

3. **HTML Report Generation**:
   ```bash
   ./checker_tool/bin/masterfabric-prerelease --check . --html security_report.html
   ```

4. **Real-time CVE API Testing** (if applicable):
   ```bash
   ./checker_tool/bin/masterfabric-prerelease --check . --verbose --cve-api
   ```

### Expected Security Findings
<!-- List what security issues should be detected or what vulnerabilities should be fixed -->

---

## 🔗 Related Security Links

- **Security Issue**: None<!-- Link to related security issue or CVE -->
- **CVE Reference**: None<!-- If applicable, link to CVE details -->
- **Security Documentation**: None<!-- Link to relevant security docs -->
- **Policy Changes**: None<!-- Link to security policy modifications -->

---

## 🔍 Security Analysis Details

### Vulnerability Categories Addressed
- [ ] Dependency vulnerabilities (CVE scanning)
- [ ] Cryptographic weaknesses
- [ ] Hardcoded secrets detection
- [ ] Binary hardening issues
- [ ] Static analysis findings
- [ ] Memory safety issues
- [ ] Concurrency vulnerabilities
- [ ] Other: <!-- Specify -->

### Security Policy Compliance
- [X] Follows `.security.yaml` configuration
- [X] Meets minimum acceptable standards
- [ ] Adheres to approved SDK requirements
- [ ] Complies with blacklisted version restrictions

Commit:
[ADD] missing strings.h header file is added to src/yaml_parser.c

Signed-off-by: Adınız <siz@e-posta.com>

gurkanfikretgunak commented:
dear @KaanaryOverFlow  missing pr rules. i will update in day pr_template then you can update be here. i ll inform u.

gurkanfikretgunak commented:
> dear @KaanaryOverFlow missing pr rules. i will update in day pr_template then you can update be here. i ll inform u.

https://github.com/masterfabric/masterfabric-pre-release/commit/d9d6f5c519248b656db787087e36b71c4b7a8ee9
⬇️
i updated. you can look for pr templete. 
you can look how to create prs with temp.  github.com/masterfabric-mobile/osmea

KaanaryOverFlow commented:
dear @gurkanfikretgunak I've updated my first comment. I think it's safe to emphasize that it is not a security vulnerability bug. While I was trying to compile the tool, I got a compilation error, and I fixed it.
